pyro/infer/importance.py

```diff
@@ -140,3 +145,78 @@ def _fn(*args, **kwargs):
     if normalized:
         log_weights = log_weights - torch.logsumexp(log_weights)
     return log_weights, model_trace, guide_trace
+
+
+def psis_diagnostic(model, guide, *args, **kwargs):
+    """
+    Computes the Pareto tail index k for a model/guide pair using the technique
+    described in [1], which builds on previous work in [2]. If :math:`0 < k < 0.5`
+    the guide is a good approximation to the model posterior, in the sense
+    described in [1]. If :math:`0.5 \\le k \\le 0.7`, the guide provides a suboptimal
+    approximation to the posterior, but may still be useful in practice. If
+    :math:`k > 0.7` the guide program provides a poor approximation to the full
+    posterior, and caution should be used when using the guide. Note, however,
+    that a guide may be a poor fit to the full posterior while still yielding
+    reasonable model predictions. If :math:`k < 0.0` the importance weights
+    corresponding to the model and guide appear to be bounded from above; this
+    would be a bizarre outcome for a guide trained via ELBO maximization. Please
+    see [1] for a more complete discussion of how the tail index k should be
+    interpreted.
+
+    Please be advised that a large number of samples may be required for an
+    accurate estimate of k.
+
+    Note that we assume that the model and guide are both vectorized and have
+    static structure. As is canonical in Pyro, the args and kwargs are passed
+    to the model and guide.
+
+    Finally note that this implementation is adapted from the implementation
+    in reference [2].
+
+    References
+    [1] 'Yes, but Did It Work?: Evaluating Variational Inference.'
+    Yuling Yao, Aki Vehtari, Daniel Simpson, Andrew Gelman
+    [2] 'Pareto Smoothed Importance Sampling.'
+    Aki Vehtari, Andrew Gelman, Jonah Gabry
+    https://github.com/avehtari/PSIS/blob/904146236767182270c9718a7b2a30831fe701fe/py/psis.py#L112
```

Comment:
Sorry, you can't adapt that code because it is GPL-3.0 licensed.

Reply:
Here's an Apache-2.0 licensed implementation of Pareto something, from arviz: https://github.com/arviz-devs/arviz/blob/master/arviz/stats/diagnostics.py#L331